src/Microsoft.AspNet.Mvc.ModelBinding/ModelBindingContext.cs

```diff
@@ -71,12 +70,14 @@ public object Model
             set
             {
                 IsModelSet = true;
-
+                ModelStateKey = ModelName;
```

Comment:
because the model name at that point in the model binding process represents the Key that was used to bind the model. 
This can be done by individual modelbinders as well but then they would need to remember one more thing to do along with setting a model value.